code/modules/antagonists/vampire/vampire_powers/dantalion_powers.dm

```diff
@@ -0,0 +1,206 @@
+/proc/isvampirethrall(mob/living/M)
+	return istype(M) && M.mind && M.mind.has_antag_datum(/datum/antagonist/mindslave/thrall)
+
+/datum/vampire_passive/increment_thrall_cap/on_apply(datum/antagonist/vampire/V)
+	V.subclass.thrall_cap += 1
+	gain_desc = "You can now thrall one more person. Up to a maximum of [V.subclass.thrall_cap]"
+
+/datum/vampire_passive/increment_thrall_cap/two
+
+/datum/vampire_passive/increment_thrall_cap/three
+
+/obj/effect/proc_holder/spell/vampire/enthrall
+	name = "Enthrall (150)"
+	desc = "You use a large portion of your power to sway those loyal to none to be loyal to you only."
+	gain_desc = "You have gained the ability to thrall people to your will."
+	action_icon_state = "vampire_enthrall"
+	required_blood = 150
+	deduct_blood_on_cast = FALSE
+
+/obj/effect/proc_holder/spell/vampire/enthrall/create_new_targeting()
+	var/datum/spell_targeting/click/T = new
+	T.range = 1
+	T.click_radius = 0
+	return T
+
+/obj/effect/proc_holder/spell/vampire/enthrall/cast(list/targets, mob/user = usr)
+	var/datum/antagonist/vampire/vampire = user.mind.has_antag_datum(/datum/antagonist/vampire)
+	var/mob/living/target = targets[1]
+	user.visible_message("<span class='warning'>[user] bites [target]'s neck!</span>", "<span class='warning'>You bite [target]'s neck and begin the flow of power.</span>")
+	to_chat(target, "<span class='warning'>You feel the tendrils of evil invade your mind.</span>")
+	if(do_mob(user, target, 15 SECONDS))
+		if(can_enthrall(user, target))
+			handle_enthrall(user, target)
+			var/datum/spell_handler/vampire/V = custom_handler
+			var/blood_cost = V.calculate_blood_cost(vampire)
+			vampire.bloodusable -= blood_cost //we take the blood after enthralling, not before
+	else
+		revert_cast(user)
+		to_chat(user, "<span class='warning'>You or your target moved.</span>")
+
+/obj/effect/proc_holder/spell/vampire/enthrall/proc/can_enthrall(mob/living/user, mob/living/carbon/C)
+	if(!C)
+		log_runtime(EXCEPTION("target was null while trying to vampire enthrall, attacker is [user] [user.key] \ref[user]"), user)
+		return FALSE
+	if(!user.mind.som)
+		stack_trace("Dantalion Thrall datum ended up null.")
+		return
```

Comment:
```suggestion
		CRASH("Dantalion Thrall datum ended up null.")
```
Will behave the same. Yet CRASH will show this proc as the origin of the runtime. Helps make runtimes a bit more clear.